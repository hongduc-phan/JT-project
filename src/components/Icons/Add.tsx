import React, {SVGAttributes} from 'react';

const Notification = (props: SVGAttributes<HTMLOrSVGElement>) => (
  <svg width="1em" height="1em" viewBox="0 0 24 24" {...props}>
    <g fill="none" fillRule="evenodd">
      <path
        fill="currentColor"
        fillRule="nonzero"
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
      />
      <path d="M0 0h24v24H0z" />
    </g>
  </svg>
);

export default Notification;
