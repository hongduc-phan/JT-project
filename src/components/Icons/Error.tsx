import React, {SVGAttributes} from 'react';

const Error = (props: SVGAttributes<HTMLOrSVGElement>) => (
  <svg width="1em" height="1em" viewBox="0 0 24 24" {...props}>
    <g fill="none" fillRule="evenodd">
      <path d="M0 0h24v24H0z" />
      <path
        fill="currentColor"
        fillRule="nonzero"
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
      />
    </g>
  </svg>
);

export default Error;
