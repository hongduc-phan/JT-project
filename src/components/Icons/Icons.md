```js
const {
  Edit,
  Delete,
  Setting,
  EPayRoll,
  ETime,
  ArrowDown,
  ArrowLeft,
  ArrowRight,
  ArrowUp,
  Notification,
  Account,
  Add,
  Search,
  Error,
} = require('./');

<div className="icon-list">
  <div>
    <div>
      <Edit />
    </div>
    <div>
      <Delete />
    </div>
    <div>
      <Setting />
    </div>
    <div>
      <Setting active={true} />
    </div>
    <div>
      <EPayRoll />
    </div>
    <div>
      <EPayRoll active={true} />
    </div>
    <div>
      <ETime />
    </div>
    <div>
      <ETime active={true} />
    </div>
    <div>
      <ArrowDown />
    </div>
    <div>
      <ArrowLeft />
    </div>
    <div>
      <ArrowUp />
    </div>
    <div>
      <ArrowRight />
    </div>
  </div>
  <div>
    <div>
      <Notification />
    </div>
    <div>
      <Account />
    </div>
    <div>
      <Add />
    </div>
    <div>
      <Search />
    </div>
    <div>
      <Error />
    </div>
  </div>
</div>;
```
