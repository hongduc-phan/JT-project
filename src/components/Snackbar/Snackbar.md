Example

```js
const Snackbar = require('./').default;
const {Error} = require('../Icons');

<div>
  <div className="snackbar-row">
    <Snackbar>Company information saved</Snackbar>
  </div>
  <div className="snackbar-row">
    <Snackbar renderIcon={<Error />} variant="error">
      Unable to save company information. Please check your internet connection.
    </Snackbar>
  </div>
</div>;
```
