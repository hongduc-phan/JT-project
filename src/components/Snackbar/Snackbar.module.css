.root {
  padding: 0.875rem 1rem;
  color: var(--white);
  box-shadow: 0 3px 5px -1px rgba(0, 0, 0, 0.2),
    0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 6px 10px 0 rgba(0, 0, 0, 0.14);
  background-color: #202020;
  border-radius: 0.25rem;
  align-items: center;
  display: inline-flex;
  min-width: 349px;
}

.error {
  background-color: #ae2626;
  color: var(--white);
}

.success {
  background-color: var(--green-dark);
  color: var(--white);
}

.icon {
  font-size: 1.5rem;
  margin-right: 0.5rem;
  display: inline-flex;
  align-self: flex-start;
}
