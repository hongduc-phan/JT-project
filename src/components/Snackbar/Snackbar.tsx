import React, {
  AllHTMLAttributes,
  Fragment,
  FunctionComponent,
  ReactNode,
} from 'react';
import clsx from 'clsx';

import Typo, {TypoVariants, TypoColors} from '../Typo';

import styles from './Snackbar.module.css';

export enum SnackbarVariant {
  Error = 'error',
  Success = 'success',
}

export interface SnackbarProps extends AllHTMLAttributes<HTMLDivElement> {
  variant?: SnackbarVariant;
  children: ReactNode;
  color?: TypoColors;
  renderIcon?: ReactNode;
}

const Snackbar: FunctionComponent<SnackbarProps> = ({
  variant,
  children,
  className,
  color,
  renderIcon,
  ...rest
}: SnackbarProps) => {
  return (
    <Typo<HTMLDivElement>
      {...rest}
      tag="div"
      variant={TypoVariants.body2}
      className={clsx(
        styles.root,
        variant === SnackbarVariant.Error && styles.error,
        variant === SnackbarVariant.Success && styles.success,
        className,
      )}
      color={color}
    >
      {renderIcon && (
        <Fragment>
          <span className={styles.icon}>{renderIcon}</span>
          <span>{children}</span>
        </Fragment>
      )}
      {!renderIcon && children}
    </Typo>
  );
};

export default Snackbar;
