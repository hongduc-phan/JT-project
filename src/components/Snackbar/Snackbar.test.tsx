import React from 'react';
import renderer from 'react-test-renderer';

import Snackbar, {SnackbarVariant} from './index';
import {Error} from '../Icons';

describe('<Snackbar />', () => {
  it('Should match snapshot normal style', () => {
    const snack = <Snackbar>Company information saved</Snackbar>;
    const tree = renderer.create(snack).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should match snapshot error style with icon', () => {
    const snack = (
      <Snackbar renderIcon={<Error />} variant={SnackbarVariant.Error}>
        Unable to save company information. Please check your internet
        connection.
      </Snackbar>
    );
    const tree = renderer.create(snack).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
