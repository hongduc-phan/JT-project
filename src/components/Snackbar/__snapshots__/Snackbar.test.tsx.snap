// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> Should match snapshot error style with icon 1`] = `
<div
  className="typo root error body2 black left"
>
  <span
    className="icon"
  >
    <svg
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h24v24H0z"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
          fill="currentColor"
          fillRule="nonzero"
        />
      </g>
    </svg>
  </span>
  <span>
    Unable to save company information. Please check your internet connection.
  </span>
</div>
`;

exports[`<Snackbar /> Should match snapshot normal style 1`] = `
<div
  className="typo root body2 black left"
>
  Company information saved
</div>
`;
